my_2d_robot/src/joint_states_subscriber.cpp

```diff
@@ -1,19 +1,150 @@
 #include "ros/ros.h"
+#include "ros/package.h"
 #include "std_msgs/String.h"
+#include <sensor_msgs/JointState.h>
+#include <string>
+#include <fstream>
+#include <sstream>
+#include <boost/filesystem.hpp>
+#include "yaml-cpp/yaml.h"
+#include <eigen3/Eigen/Dense>
 
-void joint_statesCallback(const std_msgs::String::ConstPtr& msg)
+using namespace Eigen;
+typedef Matrix<float, 2, 1> Vector2f;
+
+class SensorMeasurementData
+{
+public:
+  // Contructor.
+  SensorMeasurementData(ros::NodeHandle* n)
+  {
+    // Initialises length of links (specified in urdf).
+    link_1_ = 5.0;
```

Comment:
Values should not be hardcoded - it should work regardless of the 2d model you have in the URDF.